src/_P004_Dallas.ino

```diff
@@ -15,206 +15,305 @@
 #define PLUGIN_NAME_004       "Environment - DS18b20"
 #define PLUGIN_VALUENAME1_004 "Temperature"
 
+// OneWire commands
+#define STARTCONVO      0x44
+#define COPYSCRATCH     0x48
+#define WRITESCRATCH    0x4E
+#define CHOOSEROM       0x55
+#define READSCRATCH     0xBE
+#define READPOWERSUPPLY 0xB4
+#define RECALLSCRATCH   0xB8
+#define SKIPROM         0xCC
+#define ALARMSEARCH     0xEC
+#define BUSSEARCH       0xF0
+
+
+// Scratchpad locations
+#define TEMP_LSB        0
+#define TEMP_MSB        1
+#define HIGH_ALARM_TEMP 2
+#define LOW_ALARM_TEMP  3
+#define CONFIGURATION   4
+#define INTERNAL_BYTE   5
+#define COUNT_REMAIN    6
+#define COUNT_PER_C     7
+#define SCRATCHPAD_CRC  8
+
+// Device resolution
+#define TEMP_9_BIT  0x1F
+#define TEMP_10_BIT 0x3F
+#define TEMP_11_BIT 0x5F
+#define TEMP_12_BIT 0x7F
+
+// Device ROM
+#define ROM_DS1820_DS18S20  0x10
+#define ROM_DS1822          0x22
+#define ROM_DS18B20         0x28
+#define ROM_DS1825          0x3B
+
+// Arrays
+#define SP_SIZE 9
+#define ADDR_SIZE 8
+
 int8_t Plugin_004_DallasPin;
+boolean Plugin_004_DallasPullUp;
+static boolean Plugin_004_parasite[TASKS_MAX];
 
 boolean Plugin_004(byte function, struct EventStruct * event, String& string)
 {
-    boolean success = false;
+        boolean success = false;
 
-    switch (function)
-    {
+        switch (function)
+        {
         case PLUGIN_DEVICE_ADD:
         {
-            Device[++deviceCount].Number           = PLUGIN_ID_004;
-            Device[deviceCount].Type               = DEVICE_TYPE_SINGLE;
-            Device[deviceCount].VType              = SENSOR_TYPE_SINGLE;
-            Device[deviceCount].Ports              = 0;
-            Device[deviceCount].PullUpOption       = false;
-            Device[deviceCount].InverseLogicOption = false;
-            Device[deviceCount].FormulaOption      = true;
-            Device[deviceCount].ValueCount         = 1;
-            Device[deviceCount].SendDataOption     = true;
-            Device[deviceCount].TimerOption        = true;
-            Device[deviceCount].GlobalSyncOption   = true;
-            break;
+                Device[++deviceCount].Number           = PLUGIN_ID_004;
+                Device[deviceCount].Type               = DEVICE_TYPE_SINGLE;
+                Device[deviceCount].VType              = SENSOR_TYPE_SINGLE;
+                Device[deviceCount].Ports              = 0;
+                Device[deviceCount].PullUpOption       = true;
```

Comment:
Is the pull-up resistor of the ESP8266 enough for this?
The pull-up resistor is between 30k and 100k